packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -105,6 +105,32 @@ export class Cluster extends Resource implements ICluster {
     return new ImportedCluster(scope, id, attrs);
   }
 
+  /**
+   * Import an existing cluster to the stack from the cluster ARN.
+   */
+  public static fromClusterArn(scope: Construct, id: string, clusterArn: string): ICluster {
+    const stack = Stack.of(scope);
+    const clusterName = stack.splitArn(clusterArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName as string;
```

Comment:
Instead of this cast, let's just handle the case when `resourceName` is `undefined` (throw an exception).

See [here](https://github.com/aws/aws-cdk/blob/d735ce4bc9049b68b830641d68c97f4be565968d/packages/%40aws-cdk/aws-servicecatalog/lib/product.ts#L147-L166) for similar code in other modules.

This will also require a separate unit test.